pytorchfi/errormodels.py

```diff
@@ -0,0 +1,358 @@
+        total_bits = self.bits
+
+        quantum = int((orig_value / max_value) * ((2.0 ** (total_bits - 1))))
+        twos_comple = self._twos_complement(quantum, total_bits)  # signed
+
+        # binary representation
+        bits = bin(twos_comple)[2:]
+
+        # sign extend 0's
+        temp = "0" * (total_bits - len(bits))
+        bits = temp + bits
+        assert len(bits) == total_bits
+
+        # flip a bit
+        # use MSB -> LSB indexing
+        assert bit_pos < total_bits
+
+        bits_new = list(bits)
+        bit_loc = total_bits - bit_pos - 1
+        if bits_new[bit_loc] == "0":
+            bits_new[bit_loc] = "1"
+        else:
+            bits_new[bit_loc] = "0"
+        bits_str_new = "".join(bits_new)
+
+        # GPU contention causes a weird bug...
+        if not bits_str_new.isdigit():
+            print("orig value:", orig_value)
+            print("max value:", max_value)
+            print("bit_pos:", bit_pos)
+            print("total bits:", total_bits)
+            print("bit string:", bits_str_new)
+
+        # convert to quantum
+        assert bits_str_new.isdigit()
+        new_quantum = int(bits_str_new, 2)
+        out = self._twos_comp(new_quantum, total_bits)
+
+        # get FP equivalent from quantum
+        new_value = out * ((2.0 ** (-1 * (total_bits - 1))) * max_value)
+
+        return torch.tensor(new_value, dtype=save_type)
+
+    def single_bit_flip_signed_across_batch(self, module, input, output):
+        corrupt_conv_set = self.get_corrupt_conv()
+        range_max = self.get_conv_max(self.get_curr_conv())
+
+        if type(corrupt_conv_set) == list:
+            inj_list = list(
+                filter(
+                    lambda x: corrupt_conv_set[x] == self.get_curr_conv(),
+                    range(len(corrupt_conv_set)),
+                )
+            )
+            for i in inj_list:
+                self.assert_inj_bounds(index=i)
+                prev_value = output[self.CORRUPT_BATCH[i]][self.CORRUPT_C[i]][
+                    self.CORRUPT_H[i]
+                ][self.CORRUPT_W[i]]
+
+                rand_bit = random.randint(0, self.bits - 1)
+                new_value = self._flip_bit_signed(prev_value, range_max, rand_bit)
+
+                output[self.CORRUPT_BATCH[i]][self.CORRUPT_C[i]][self.CORRUPT_H[i]][
+                    self.CORRUPT_W[i]
+                ] = new_value
+
+        else:
+            self.assert_inj_bounds()
+            if self.get_curr_conv() == corrupt_conv_set:
+                prev_value = output[self.CORRUPT_BATCH][self.CORRUPT_C][self.CORRUPT_H][
+                    self.CORRUPT_W
+                ]
+
+                rand_bit = random.randint(0, self.bits - 1)
+                new_value = self._flip_bit_signed(prev_value, range_max, rand_bit)
+
+                output[self.CORRUPT_BATCH][self.CORRUPT_C][self.CORRUPT_H][
+                    self.CORRUPT_W
+                ] = new_value
+
+        self.updateConv()
```

Comment:
I think that we can simplify this and remove the `updateConv()` function and instead just put in `self.CURRENT_CONV += 1`

Reply:
Same for other instances

Reply:
I kept updateConv(), because it a pretty important step in making a new error model. Technically, it just calls `self.CURRENT_CONV += 1` underneath the hood, but encapsulating it as a function call I think is useful. Plus, I foresee other use cases where the user might want to `updateConv()`  not at the end of a function, and it might get lost as a `+=1` construct.

Reply:
I removed other functions though with simplistic returns. If needed, they can be added in the future.

Reply:
good point. made a few changes to adhere to this.